classes/tables/entries_table.php

```diff
@@ -27,11 +27,19 @@ class entries_table extends table_sql {
 
     public $cmid;
 
-    public function __construct($cmid, $cpdlogbookid, $userid, $uniqueid) {
+    public function __construct($cmid, $userid, $uniqueid) {
+        global $DB;
+
         parent::__construct($uniqueid);
         $this->cmid = $cmid;
 
-        $this->set_sql('*', '{cpdlogbook_entries}', 'cpdlogbookid=? AND userid=?', [$cpdlogbookid, $userid]);
+        list ($course, $cm) = get_course_and_cm_from_cmid($cmid, 'cpdlogbook');
+
+        require_course_login($course, false, $cm);
```

Comment:
I would avoid having this side effect inside the constructor, you could, in theory, want to use this table in a scenario which doesn't even have a login such as a cron job. In this case I think it redundant as you call it in view.php

I'd try to keep this constructor as light as possible as most of this stuff should have already been done on the pages that call it and things can be optionally passed in instead, eg pass in the cm instance id instead of the cmid